### PR TITLE
Add module test run configuration support based on given configuratio…

### PR DESCRIPTION
**Pull request by MichalBednarz**
…n display name fragment

### CODE REVIEW

File: extensions/scala/probePlugin/src/main/scala/org/virtuslab/ideprobe/scala/ScalaTestRunConfigurationTransformer.scala [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import org.jetbrains.plugins.scala.testingSupport.test.scalatest.ScalaTestRunConfiguration
+import org.virtuslab.ideprobe.handlers.RunConfigurationTransformer
+
+class ScalaTestRunConfigurationTransformer extends RunConfigurationTransformer {
+  override def transform(runConfiguration: RunnerAndConfigurationSettings): RunnerAndConfigurationSettings = {
+    runConfiguration.getConfiguration match {
+      case scalaConfig: ScalaTestRunConfiguration => {
+        scalaConfig.testConfigurationData.setUseSbt(true)
+        scalaConfig.testConfigurationData.setUseUiWithSbt(true)
+        runConfiguration.asInstanceOf[RunnerAndConfigurationSettingsImpl].setConfiguration(scalaConfig)
```
Comment (lukaszwawrzyk) [inline extensions/scala/probePlugin/src/main/scala/org/virtuslab/ideprobe/scala/ScalaTestRunConfigurationTransformer.scala:14]: aren't lines above mutating the state anyway?

File: probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala [hunk 2]
```diff
@@ -1,37 +1,89 @@
...
+
+      val dataContext = new MapDataContext
+      dataContext.put(CommonDataKeys.PROJECT, project)
+      dataContext.put(LangDataKeys.MODULE, module)
+
+      val projectPath = Paths.get(project.getBasePath)
+
+      // TODO getContentRoots returns an array, is taking head only reliable?
+      val moduleVirtualFile = ModuleRootManager.getInstance(module).getContentRoots.head
+      val moduleDir = runOnUISync {
```
Comment (lukaszwawrzyk) [inline probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala:43]: It sounds like `read` should do here

File: probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala [hunk 3]
```diff
@@ -1,37 +1,89 @@
...
+
+      val wsVirtualFile = VFS.toVirtualFile(projectPath)
+      dataContext.put(PlatformDataKeys.PROJECT_FILE_DIRECTORY, wsVirtualFile)
+
+      val location = PsiLocation.fromPsiElement(moduleDir)
+      dataContext.put(Location.DATA_KEY, location)
+
+      val configurationContext = ConfigurationContext.getFromContext(dataContext)
+      val runManager = configurationContext.getRunManager.asInstanceOf[RunManagerEx]
+      val configurationFromContext = runOnUISync(configurationContext.getConfiguration)
```
Comment (lukaszwawrzyk) [inline probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala:57]: just `read`

File: probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala [hunk 4]
```diff
@@ -1,37 +1,89 @@
...
+
+      runManager.setTemporaryConfiguration(configurationFromContext)
+      runManager.setSelectedConfiguration(configurationFromContext)
+
+      val configurations = runOnUISync {
```
Comment (lukaszwawrzyk) [inline probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala:62]: just `read`

File: probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala [hunk 5]
```diff
@@ -1,37 +1,89 @@
...
+        configurationContext.getConfigurationsFromContext
+      }
+      val producer = runConfiguration.runnerNameFragment match {
+        case Some(fragment) =>
+          configurations
+            .find(_.toString.toLowerCase contains fragment.toLowerCase)
```
Comment (lukaszwawrzyk) [inline probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala:68]: let's make it case sensitive
do we need substring? what are the example names?

File: probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala [hunk 6]
```diff
@@ -1,37 +1,89 @@
...
+      val runManager = configurationContext.getRunManager.asInstanceOf[RunManagerEx]
+      val configurationFromContext = runOnUISync(configurationContext.getConfiguration)
+
+      runManager.setTemporaryConfiguration(configurationFromContext)
+      runManager.setSelectedConfiguration(configurationFromContext)
+
+      val configurations = runOnUISync {
+        configurationContext.getConfigurationsFromContext
+      }
+      val producer = runConfiguration.runnerNameFragment match {
```
Comment (lukaszwawrzyk) [inline probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala:65]: maybe just `name`? for now

File: probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala [hunk 7]
```diff
@@ -1,37 +1,89 @@
...
+          configurations
+            .find(_.toString.toLowerCase contains fragment.toLowerCase)
+            .getOrElse(
+              throw new RuntimeException(
+                s"Runner name fragment $fragment does not match any configuration name. Available configurations: $configurations."
+              )
+            )
+        case _ =>
+          configurations.headOption.getOrElse(
+            throw new RuntimeException(
```
Comment (lukaszwawrzyk) [inline probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala:76]: just use `error` method from IntelliJApi

**Review by lukaszwawrzyk — APPROVED** [review]

File: probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala (hunk 5 above)
Comment (MichalBednarz) [reply probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala:68]: the examples are `JUnit: All in Hello (1) (level: WORKSPACE)` or `ScalaTest: ScalaTests in 'ws' (level: WORKSPACE)` so provider runner names like `JUnit` or `ScalaTest` could be applied in such cases

File: extensions/scala/probePlugin/src/main/scala/org/virtuslab/ideprobe/scala/ScalaTestRunConfigurationTransformer.scala (hunk 1 above)
Comment (MichalBednarz) [reply extensions/scala/probePlugin/src/main/scala/org/virtuslab/ideprobe/scala/ScalaTestRunConfigurationTransformer.scala:14]: indeed

**Pull request merged**